waitgroup.go

```diff
@@ -41,3 +41,12 @@ func (h *WaitGroup) Wait() {
 	// Propagate a panic if we caught one from a child goroutine.
 	h.pc.Repanic()
 }
+
+// WaitSafe will block until all goroutines spawned with Go exit and will
+// return a *panics.RecoveredPanic if one of the child goroutines panics.
+func (h *WaitGroup) WaitSafe() *panics.RecoveredPanic {
```

Comment:
I'm starting to think that `WaitAndRecover()` might be a better name. The term "Safe" is pretty overloaded, and it's not super clear what it means without reading the doc string. "Recover" is a much more specific term regarding panics, and is a little more intuitive from my perspective. 

The extra verbosity seems acceptable given that most users should probably be using `Wait()` still.

What do you think?

Reply:
I think that is a good call. Updated. Thanks!